exercise/templatetags/exercise.py

```diff
@@ -167,16 +161,16 @@ def get_points_class(context):
         'not_passed': 'badge-danger',
         'missing': 'badge-danger',
     }
-    class_name =  options.get(points_type[0], '')
-    if len(points_type) > 1 and points_type[1] == "unconfirmed":
+    class_name = options.get(next(filter(lambda x: x != "unconfirmed", points_type), ""), "")
+    if "unconfirmed" in points_type:
         class_name += " unconfirmed-points"
     return class_name
 
 # Adds the content of the points_badge.
 @register.simple_tag(takes_context=True)
 def get_points_text(context):
     points_string = str(context['points']) + " / " + str(context['max'])
-    points_type = _get_points_type(context)[0]
+    points_type = next(filter(lambda x: x != "unconfirmed", _get_points_type(context)), "")
```

Comment:
As we can see, set doesn't after all work as well as predicted, even though it would be nice in some similar cases. So, maybe just return tuple of two arguments? That is also trivial to unpack.

so `return (class_, unconfirmed)` and then `points_type, unconfirmed = _get_points_type(context)[0]`

Reply:
Done.